gcloud/_helpers.py

```diff
@@ -388,6 +389,43 @@ def _datetime_to_pb_timestamp(when):
     return timestamp_pb2.Timestamp(seconds=seconds, nanos=nanos)
 
 
+def _name_from_project_path(path, project, template):
+    """Validate a URI path and get the leaf object's name.
+
+    :type path: string
+    :param path: URI path containing the name.
+
+    :type project: string
+    :param project: The project associated with the request. It is
+                    included for validation purposes.
+
+    :type template: string
+    :param template: Template regex describing the expected form of the path.
+                     The regex must have two named groups, 'project' and
+                     'name'.
+
+    :rtype: string
+    :returns: Name parsed from ``path``.
+    :raises: :class:`ValueError` if the ``path`` is ill-formed or if
+             the project from the ``path`` does not agree with the
+             ``project`` passed in.
+    """
+    if isinstance(template, str):
+        template = re.compile(template)
```

Comment:
They are *always* going to be native `str`, because they are created in code as string literals.  We don't need to worry about Unicode, because nobody is going to be using them to parse API-centric paths.